### ISSUE TITLE
Tests to check if IDGraph can correctly detect modifications for widely used data science and machine learning libraries.

### ISSUE THREAD

**Issue by Ribhavs2** [illinoisdata/kishu#89]
We want to test (1) ID-Graph and (2) end-to-end testing against the following libraries:

- [x] pandas
  - [x] series
  - [x] dataframe
- [x] numpy
- [x] seaborn
  - [x] displot
  - [x] scatterplot
  - [x] barplot
- [x] matplotlib.pyplot
  - [x] AxesSubplot
  - [x] Figure
- [ ] sklearn
  - [ ] sklearn.linear_model
  - [ ] sklearn.metrics
  - [ ] sklearn.model_selection
  - [ ] sklearn.cluster
  - [ ] sklearn.preprocessing
  - [ ] sklearn.ensemble
- [ ] scipy.stats
- [ ] lightgbm
- [ ] xgboost
- [ ] keras
  - [ ] keras.models
  - [ ] keras.layers
  - [ ] keras.optimizers
- [ ] tensorflow
- [ ] cv2
- [ ] torch
- [ ] torchvision

Links for notebooks from which the above libraries were implemented:

<html>
<body>
<google-sheets-html-origin>

https://www.kaggle.com/code/pmarcelino/comprehensive-data-exploration-with-python
https://www.kaggle.com/code/serigne/stacked-regressions-top-4-on-leaderboard
https://www.kaggle.com/code/yassineghouzam/introduction-to-cnn-keras-0-997-top-6
https://www.kaggle.com/code/ldfreeman3/a-data-science-framework-to-achieve-99-accuracy
https://www.kaggle.com/code/willkoehrsen/start-here-a-gentle-introduction
https://www.kaggle.com/code/janiobachmann/credit-fraud-dealing-with-imbalanced-datasets
https://www.kaggle.com/code/dansbecker/your-first-machine-learning-model
https://www.kaggle.com/code/its7171/nfl-baseline-simple-helmet-mapping
https://www.kaggle.com/code/startupsci/titanic-data-science-solutions
https://www.kaggle.com/code/helgejo/an-interactive-data-science-tutorial
https://www.kaggle.com/code/philculliton/inferring-birds-with-kaggle-models
https://www.kaggle.com/code/abhishek/approaching-almost-any-nlp-problem-on-kaggle
https://www.kaggle.com/code/pestipeti/pytorch-starter-fasterrcnn-train
https://matplotlib.org/stable/api/figure_api.html#matplotlib.figure.Figure.add_subplot
https://matplotlib.org/stable/users/explain/figures.html#figure-explanation

</body>
</html>
